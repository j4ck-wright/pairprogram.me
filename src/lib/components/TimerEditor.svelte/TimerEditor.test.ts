import { fireEvent, render, screen } from '@testing-library/svelte';
import { TimerEditor } from '..';

describe('Timer Editor Test Suite', () => {
	beforeEach(() => {
		render(TimerEditor);
	});
	it('Should correctly render the timer editor with + and 15 visible', () => {
		expect(screen.getByText('+')).toBeTruthy();
		expect(screen.getByText('-')).toBeTruthy();
		expect(screen.findByText(10)).toBeTruthy();
	});

	it('Should correctly increment the timer when the + button is clicked', async () => {
		const incrementBtn = screen.getAllByRole('button')[0];
		await fireEvent.click(incrementBtn);
		expect(screen.findByText(11)).toBeTruthy();
	});

	it('Should correctly decrease the timer when the + button is clicked', async () => {
		const decreaseButton = screen.getAllByRole('button')[0];
		await fireEvent.click(decreaseButton);
		expect(screen.findByText(9)).toBeTruthy();
	});

	it('Should correctly set the timer to not go below 1', async () => {
		const decreaseButton = screen.getAllByRole('button')[0];
		for (let i = 0; i < 10; i++) {
			await fireEvent.click(decreaseButton);
		}
		expect(screen.findByText(1)).toBeTruthy();
	});
});

describe('ParticipantBadge snapshot', () => {
	it('Should match snapshot', () => {
		const participantComponent = render(TimerEditor);
		expect(participantComponent).toMatchSnapshot();
	});
});
