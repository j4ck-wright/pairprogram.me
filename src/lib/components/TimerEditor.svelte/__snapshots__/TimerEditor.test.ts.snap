// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ParticipantBadge snapshot > Should match snapshot 1`] = `
{
  "component": TimerEditor {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        10,
        [Function],
        [Function],
        [Function],
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {},
      "root": <div>
        <div
          class="container mt-2"
        >
          <span
            class="mr-12"
          >
            Timer:
          </span>
           
          <div
            class="join"
          >
            <button
              class="btn btn-xs no-animation btn-primary text-white join-item rounded-l-full"
            >
              -
            </button>
             
            <input
              class="w-12 bg-transparent join-item text-center outline-primary svelte-14nvfch"
              data-testid="timerBox"
              type="number"
            />
             
            <button
              class="btn btn-xs no-animation btn-primary text-white join-item rounded-r-full"
            >
              +
            </button>
          </div>
           
          <span
            class="ml-2"
          >
            Minutes
          </span>
        </div>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$events_def": undefined,
    "$$prop_def": undefined,
    "$$set": undefined,
    "$$slot_def": undefined,
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <div
        class="container mt-2"
      >
        <span
          class="mr-12"
        >
          Timer:
        </span>
         
        <div
          class="join"
        >
          <button
            class="btn btn-xs no-animation btn-primary text-white join-item rounded-l-full"
          >
            -
          </button>
           
          <input
            class="w-12 bg-transparent join-item text-center outline-primary svelte-14nvfch"
            data-testid="timerBox"
            type="number"
          />
           
          <button
            class="btn btn-xs no-animation btn-primary text-white join-item rounded-r-full"
          >
            +
          </button>
        </div>
         
        <span
          class="ml-2"
        >
          Minutes
        </span>
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
